### PR TITLE
Dev compress refactor

### PR DESCRIPTION
**Pull request by QuanluZhang**

### CODE REVIEW

File: docs/en_US/Compressor/AutoCompression.md [hunk 1]
```diff
@@ -9,13 +9,13 @@ You can easily compress a model with NNI compression. Take pruning for example,
 ```python
 from nni.compression.torch import LevelPruner
 config_list = [{ 'sparsity': 0.8, 'op_types': ['default'] }]
-pruner = LevelPruner(config_list)
-pruner(model)
+pruner = LevelPruner(model, config_list)
```
Comment (liuzhe-lz) [inline docs/en_US/Compressor/AutoCompression.md]: Logically a pruner should be a standalone component. It should not need a model to "create" a pruner. So this looks strange.
But I known we don't have a better solution yet.

File: docs/en_US/Compressor/Overview.md [hunk 2]
```diff
@@ -111,20 +111,26 @@ If you want to write a new pruning algorithm, you can write a class that inherit
...
+        """
+        Suggest you to use the NNI defined spec for config
+        """
+        super().__init__(model, config_list)
+
+    def calc_mask(self, layer, config):
+        """
+        Pruners should overload this method to provide mask for weight tensors.
+        The mask must have the same shape and type comparing to the weight.
+        It will be applied with `mul()` operation on the weight.
```
Comment (liuzhe-lz) [inline docs/en_US/Compressor/Overview.md]: numpy/pandas use ` ``inline code`` `

File: docs/en_US/Compressor/Overview.md [hunk 3]
```diff
@@ -111,20 +111,26 @@ If you want to write a new pruning algorithm, you can write a class that inherit
...
+        This method is effectively hooked to `forward()` method of the model.
+
+        Parameters
+        ----------
+        layer: LayerInfo
```
Comment (liuzhe-lz) [inline docs/en_US/Compressor/Overview.md]: There should be a space before colon. My bad.

File: docs/en_US/Compressor/AutoCompression.md (hunk 1 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/AutoCompression.md]: yes, let's keep it for now

File: docs/en_US/Compressor/Overview.md (hunk 2 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Overview.md]: fixed, thanks

File: docs/en_US/Compressor/Overview.md (hunk 3 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Overview.md]: fixed, thanks

**Pull request merged**